### ISSUE TITLE
ShipOrder not working

### ISSUE THREAD

**Issue by jpnathannew** [fillup/walmart-partner-api-sdk-php#22]
Dear team, 

I tried with ShipOrder api call. But when I call the api I got the 

`( ! ) Fatal error: Uncaught exception 'GuzzleHttp\Exception\ClientException' with message 'Client error response [url] https://marketplace.walmartapis.com/v3/orders/1576859875606/shipping [status code] 400 [reason phrase] Bad Request' in C:\wamp\www\project\vendors\guzzlehttp\command\src\AbstractClient.php on line 171`

When i print the request XML Everything looks good

```
<?xml version="1.0" encoding="UTF-8"?>
<ns3:orderShipment xmlns:ns3="http://walmart.com/mp/v3/orders">
    <ns3:orderLines>
        <ns3:orderLine>
            <ns3:lineNumber>1</ns3:lineNumber>
            <ns3:orderLineStatuses>
                <ns3:orderLineStatus>
                    <ns3:status>Shipped</ns3:status>
                    <ns3:statusQuantity>
                        <ns3:unitOfMeasurement>Each</ns3:unitOfMeasurement>
                        <ns3:amount>1</ns3:amount>
                    </ns3:statusQuantity>
                    <ns3:trackingInfo>
                        <ns3:shipDateTime></ns3:shipDateTime>
                        <ns3:carrierName>
                            <ns3:carrier>UPS</ns3:carrier>
                        </ns3:carrierName>
                        <ns3:methodCode>Standard</ns3:methodCode>
                        <ns3:trackingNumber>1Z12345E1392654435</ns3:trackingNumber>
                        <ns3:trackingURL></ns3:trackingURL>
                    </ns3:trackingInfo>
                </ns3:orderLineStatus>
            </ns3:orderLineStatuses>
        </ns3:orderLine>
    </ns3:orderLines>
</ns3:orderShipment>
```

Please help me to find the issue

**Comment by jpnathannew** [comment]
When I check the walmart api documentation I detect the issue on our code

[https://developer.walmartapis.com/#shipping-notificationsupdates](https://developer.walmartapis.com/#shipping-notificationsupdates)

I found example xml have `ns2` name space. But when I print the request xml its shows `ns2`.

After that I check the walmart-partner-api-sdk-php\src\orders.php file, **_ship_** function

```
$schema = [
            '/orderShipment' => [
                'namespace' => 'ns3',
                'childNamespace' => 'ns3',
            ],
            '/orderShipment/orderLines' => [
                'sendItemsAs' => 'orderLine',
            ],
            '/orderShipment/orderLines/orderLine/orderLineStatuses' => [
                'sendItemsAs' => 'orderLineStatus',
            ],
            '@namespaces' => [
                'ns3' => 'http://walmart.com/mp/v3/orders'
            ],
        ];
```

So in this function I will change all **ns3** to **ns2**, and then added one more `xmlns` (Get it from documentation).

Its working fine now. So Can you please update your code or please suggest me something.

**Comment by fillup** [comment]
I think the documentation is incorrect actually. All the other `v3` order related apis use `ns3`, so I think they just missed it in the docs for this API. You could try changing this locally in your code to see if it makes a difference, otherwise you'll have to do some debugging to get the full error response for more details as to why it failed. Let me know what you find.

**Comment by rmarriott-db** [comment]
I've tested shipping notifications, and they're working for me. However, a couple things you might want to consider/try. In the API documentation, it says the tracking URL isn't required. However, during the testing phase Walmart insisted we have a valid tracking URL in there. So, you might want to try putting a value in. If not, you might want to try removing the element altogether. Early on in my testing I found that blank fields/empty elements caused XSD validation errors, which caused Walmart to reject my uploads.

Last thing to note is that the Walmart Seller Center offers an "Other" carrier option when manually marking orders as shipped. However, CommonComponentsV3.xsd doesn't include Other or carriers like DHL in the list of known carriers, so if you use a something else, you'll have to do something like the following:

```
<ns2:otherCarrier> 
<ns2:carrier>DHL</ns2:carrier> 
</ns2:otherCarrier>
```

However, your example above is using UPS, so this might not apply to you.

**Comment by fillup** [comment]
Thanks for the input @rmarriott-db, does that help you @jpnathannew?

**Comment by rmarriott-db** [comment]
Just a correction to my post above. If you're using a different carrier than the ones Walmart specifies in their XSD file, the format of your XML should look like the following (not what I posted above):

```
<ns2:carrierName> 
<ns2:otherCarrier>DHL</ns2:otherCarrier> 
</ns2:carrierName>
```

So, you would essentially build your array like this:

```
...
'trackingInfo' => [
    'shipDateTime' => $shipDateTime,
    'carrierName' => [
        'otherCarrier' => $carrier
    ]
...
```

It took a some discussions with the Walmart API support team before I got the format right. Hopefully that will save someone some time, as there are many more carrier options out there than Walmart anticipated in their list.
